internal/transaction/rm.go

```diff
@@ -0,0 +1,19 @@
+// MIT License. Copyright (c) 2020 CQFN
+// https://github.com/cqfn/degitx/blob/master/LICENSE
+
+package transaction
+
+import "context"
+
+// ResourceManager API
+type ResourceManager interface {
```

Comment:
From `Manager` interface docs follows that `RM` starts a transaction. So it means that `RM` has to have some interfaces for that, isn't it?
It's unclear where is entry point to entire update workflow, while `Manager` also has no methods to receive data from clients, or for now it's out of scope?

Reply:
You're right, it seems the `Vote` method is not a pubilc contract of the RM. Also, I've renamed it `ResourceManager` -> `Resource` to avoid confision in `transaction` package (`transaction.Manager` and `transaction.ResourceManager`).

These interfaces `Manager` and `Resource` defines the public contract for components: when the RM receives a prepared message from hook and successfully broadcasted the vote to acceptors, it begin the transaction by calling public API of the TM using `Manager` interface; the TM, when it's able to make a decision, calls RM public API with `Resource` interface with `Abort()` or `Commit()` decisions; then, the RM interact with a hook and notify the TM using `Finish()` call of public interface when the git reference transaction completed.